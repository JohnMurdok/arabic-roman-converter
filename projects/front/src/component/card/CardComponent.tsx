import * as React from 'react';
import Box from '@mui/material/Box';
import Card from '@mui/material/Card';
import Typography from '@mui/material/Typography';
import CardContent from '@mui/material/CardContent';

interface ICardComponentProps {
  title: string;
  children: React.ReactNode | React.ReactNode[]
}

const CardComponent = (props: ICardComponentProps) => (
  <Box data-testid={'card-component'} sx={{ minWidth: 275, marginTop: 25 }}>
    <Card variant="outlined">
      <CardContent data-testid={'card-component-content'}>
        <Typography sx={{ fontSize: 14 }} color="text.secondary" gutterBottom>
          { props.title }
        </Typography>
        { props.children }
      </CardContent>
    </Card>
  </Box>
);

export default CardComponent;
