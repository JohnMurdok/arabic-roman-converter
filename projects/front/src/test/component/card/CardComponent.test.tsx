import { render, screen } from '@testing-library/react';
import CardComponent from '../../../component/card/CardComponent';
import fixtures from './fixtures';

describe('[PAGE] CardComponent', () => {
    test('matches snapshot', async () => {
        render(<CardComponent {...fixtures} />);
        const cardComponent = await screen.findByTestId('card-component');
        expect(cardComponent).toMatchSnapshot()
    });
});
