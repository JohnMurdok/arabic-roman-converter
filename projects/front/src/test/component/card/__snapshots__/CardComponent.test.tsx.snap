// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[PAGE] CardComponent matches snapshot 1`] = `
<div
  class="MuiBox-root css-yat5fy"
  data-testid="card-component"
>
  <div
    class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1v5z4dq-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      data-testid="card-component-content"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
      >
        test card
      </p>
      <div>
        test1
      </div>
    </div>
  </div>
</div>
`;
